plugins/optimization-detective/storage/class-od-rest-url-metrics-store-controller.php

```diff
@@ -0,0 +1,359 @@
+<?php
+/**
+ * REST API integration for the plugin: OD_REST_URL_Metrics_Store_Controller.
+ *
+ * @package optimization-detective
+ * @since 0.1.0
+ */
+
+// @codeCoverageIgnoreStart
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+// @codeCoverageIgnoreEnd
+
+/**
+ * OD_REST_URL_Metrics_Store_Controller class
+ *
+ * @since n.e.x.t
+ *
+ * @access private
+ */
+class OD_REST_URL_Metrics_Store_Controller {
```

Comment:
The class implementation now is almost entirely static, so that's contrary to what was discussed on the PR before, I think.

Per https://github.com/WordPress/performance/pull/1865#issuecomment-2701757708, we should rather call this class `OD_REST_URL_Metrics_Store_Endpoint`, and registration should not be part of the class, instead it should have a getter method like `get_registration_args()`.